src/lib.rs

```diff
@@ -735,6 +735,10 @@ where
         self.control.config.dbtr = btr;
 
         let can = self.registers();
+        if btr.tdc() {
+            let tcdo = btr.dtseg1() * btr.dbrp();
+            can.tdcr.write(|w| unsafe { w.tdco().bits(tcdo) });
```

Comment:
`dtseg1` and `dbrp` are both in the range 1 - 32 inclusive. Should there be a ceiling here so that the value doesn't overflow the 7-bit `TDCO` register? There's a hint in this section of RM0440
![image](https://github.com/user-attachments/assets/38a3da15-ab66-4196-8542-da963a742e6c)

Also I think the value written to the register should be `tcdo - 1`